apps/homescreen/js/pages.js

```diff
@@ -1,37 +1,77 @@
-/* global PagesStore, IconsHelper */
+/* global PagesMetadata, PagesStore, IconsHelper, Settings */
 'use strict';
 
 (function(exports) {
-
   /**
    * The size in pixels of the icons in the pinned pages.
    */
   const PAGES_ICON_SIZE = 30;
 
+  /**
+   * Timeout before resizing the apps grid after apps change.
+   */
+  const RESIZE_TIMEOUT = 500;
+
+  /**
+   * The distance at the top and bottom of the icon container that when hovering
+   * an icon in will cause scrolling.
+   */
+  const AUTOSCROLL_DISTANCE = 40;
+
+  /**
+   * The timeout before auto-scrolling a page when hovering at the edges
+   * of the grid.
+   */
+  const AUTOSCROLL_DELAY = 750;
+
+  /**
+   * The time to wait after setting a scroll-position before disabling
+   * overflow during drag-and-drop.
+   */
+  const AUTOSCROLL_OVERFLOW_DELAY = 500;
+
   function Pages() {
     // Element references
     this.panel = document.getElementById('pages-panel');
-    this.panels = document.getElementById('panels');
     this.pages = document.getElementById('pages');
     this.scrollable = document.querySelector('#pages-panel > .scrollable');
-    this.bottombar = document.getElementById('bottombar');
     this.remove = document.getElementById('remove');
     this.done = document.getElementById('done');
 
     // Tracking if the list is empty
     this.empty = true;
 
+    // Paging
+    this.resizeTimeout = null;
+    this.pageHeight = 1;
+    this.pendingGridHeight = 1;
+    this.iconsLeft = 0;
+    this.iconsRight = 0;
+
+    // Drag-and-drop
+    this.dragging = false;
+    this.draggedIndex = -1;
+    this.autoScrollInterval = null;
+    this.autoScrollOverflowTimeout = null;
+    this.hoverIcon = null;
+
     // Edit mode
     this.editMode = false;
 
-    // Dialogs
-    this.dialogs = [];
```

Comment:
I'd rather this change wasn't made right now - we have a bug open to add a confirm dialog for deleting pinned pages and we'll just have to add this back in again (but it could easily be forgotten).